Add docker ansible playbook

chetnieter opened the pull request:
Added the installation of docker and the pulling of my initial Docker image for the nwchem post-processing to ansible playbook.

chetnieter committed:
Added docker to ansible playbook.

Added docker to ansible playbook for the VM, add the hpccloud user
to the docker group and pull my initial docker image with the
nwchem json conversion script.

chetnieter committed:
Added newline to end of docker yaml file.

chetnieter commented:
@cjh1 PTL

chetnieter commented:
@cjh1 Should I change hosts for the docker set up in site.yml to nwchem and then add a nwchem group [here](https://github.com/Kitware/HPCCloud-deploy/blob/master/Vagrantfile#L96-L99)?

cjh1 commented:
I think docker will be used by more the just nwchem, so the group should probably be called docker.

chetnieter committed:
Added a docker group to the ansible provisioning.

I added a new docker group to the ansible provisioning. This will
allow better control over provisioning in cases where docker is not
needed.

cjh1 commented on ansible/roles/docker/tasks/main.yml:
```diff
@@ -0,0 +1,13 @@
+- name: Install docker
+  become_user: root
```
```become_user``` should be paired with ```become: yes``` ( to ensure privilege escalation is activated )

chetnieter replied:
Does this apply each task with become_user or just the install one? It doesn't seem to be consistent for the other roles.

cjh1 replied:
Any that use become_user.
